File: ios/practice-journal-ios/practice-journal-ios/Models/PracticeSession.swift
import SwiftUI

struct PracticeSession: Codable {
    let id: UUID
    let userId: String
    let duration: TimeInterval
    let notes: String
    let date: Date
    let recordingPath: String?
    
    init(userId: String, 
         duration: TimeInterval, 
         notes: String, 
         recordingPath: String? = nil) {
        self.id = UUID()
        self.userId = userId
        self.duration = duration
        self.notes = notes
        self.date = Date()
        self.recordingPath = recordingPath
    }
}

File: backend/models/PracticeSession.js
const mongoose = require("mongoose");

const practiceSessionSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true,
  },
  duration: {
    type: Number, // TimeInterval will be stored as Number
    required: true,
  },
  notes: {
    type: String,
    default: "",
  },
  date: {
    type: Date,
    default: Date.now,
  },
  recordingPath: {
    type: String,
    default: null,
  },
});

module.exports = mongoose.model("PracticeSession", practiceSessionSchema);
